forem/forem
# Refactor reCaptcha code and bypass in report abuse for trustworthy users only

## What type of PR is this? (check all applicable)


## Description

There's been plenty of reports that the reCaptcha in the abuse report form is problematic. This is usually the case in certain scenarios, i.e. users that use a VPN usually are "non-trustworthy" from Google's standpoint so it becomes very difficult to get these reports in (see the discussion in #9919).

**After leaving this PR in the ice box for a few weeks it still does the same thing but using a different implementation approach:**

Due to ReCaptcha being used in the Registration Form this PR refactors reCaptcha logic into it's own service 
- Anywhere else reCaptcha wants to be used it can just rely on using a thoroughly tested service
- It helps avoid spreading out reCaptcha code throughout controllers, helpers and duplicated tests
- For this I used #11265 (service generator)

## Related Tickets & Documents

#9919
#10283

## QA Instructions, Screenshots, Recordings

1. When logged in with a user that hasn't received a vomit reaction (at least 1 month old) you should be able to fill out the report abuse form without the reCaptcha requirement.
1. Unauthenticated users will always have to complete the reCaptcha test

## Added tests?

- [x] no, because I adapted the existing tests

## Added to documentation?


## [optional] What gif best describes this PR or how it makes you feel?

![these arent the droids youre looking for](https://media.giphy.com/media/kqYd0cqg8ihq0/giphy.gif)
